jobrunner/create_or_update_jobs.py

```diff
@@ -107,11 +113,11 @@ def create_jobs(job_request):
 def create_jobs_with_project_file(job_request, project_file):
     project = parse_and_validate_project_file(project_file)
 
-    active_jobs = get_active_jobs_for_workpace(job_request.workspace)
-    new_jobs = get_jobs_to_run(job_request, project, active_jobs)
+    current_jobs = get_latest_job_for_each_action(job_request.workspace)
+    new_jobs = get_jobs_to_run(job_request, project, current_jobs)
 
     if not new_jobs:
-        if active_jobs:
+        if any(job.state in [State.PENDING, State.RUNNING] for job in current_jobs):
             raise JobRequestError("All requested actions were already scheduled to run")
```

Comment:
Is this error message correct?  It talks about "all requested actions", but the condition uses `any()`.

Reply:
It is correct, but I agree that the logic isn't particularly clear here partly, I think, because we're trying to mirror behaviour which was implemented a long time ago and could probably be made more sensible. I've tried to restructure this to make it a bit less confusing though.